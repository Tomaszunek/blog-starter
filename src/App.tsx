import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import './App.css';


export namespace App {
  export interface IProps extends RouteComponentProps<void> {
    numbers: any   
  }
}

export default class App extends React.Component<App.IProps> {    
  public render() {
    return (
      <div className="App">
        abc
      </div>
    );
  }
}


