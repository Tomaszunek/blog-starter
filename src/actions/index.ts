export * from './articles';
export * from './products';
export * from './mainpage';
export * from './abilities';
export * from './projects';
export * from './appSystem';

