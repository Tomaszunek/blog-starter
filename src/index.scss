@import 'src/styles/variables.scss';

body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
  background-color: $body-background;
}

.container {
  padding: 0px 100px;
  background-color: $primary-background;

  @media only screen and (min-width: $sm) and (max-width: $lg) {
    padding: 0px 80px;
  }
  @media only screen and (min-width: $xs) and (max-width: $sm) {
    padding: 0px 60px;
  }
  @media only screen and (max-width: $xs) {
    padding: 0px 40px;ab
  }   
}