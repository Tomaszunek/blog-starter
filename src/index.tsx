import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Application } from './router';
import { configureStore } from './store';
import { createBrowserHistory } from 'history';
import registerServiceWorker from './registerServiceWorker';
import { Provider } from 'react-redux';
import { Router, Switch } from 'react-router';
import Navigation from 'src/pages/Navigation';
import Footer from 'src/pages/Footer';
import './index.scss';

const store = configureStore();
const history = createBrowserHistory();

ReactDOM.render(
  <Provider store={store}>    
    <Router history={history}>
      <div>
        <Navigation/>
          <Switch>
            <Application />
          </Switch>
        <Footer/>
      </div>
    </Router>   
  </Provider>,  
  document.getElementById('root') as HTMLElement
);
registerServiceWorker();