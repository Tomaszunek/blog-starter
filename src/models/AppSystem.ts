export interface IAppSystem {
    loading: boolean   
} 