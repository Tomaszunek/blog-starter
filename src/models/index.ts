export * from './ArticleModel';
export * from './ProductModel';
export * from './ContentCategory';
export * from './MainPageContentModel';
export * from './AppSystem';

