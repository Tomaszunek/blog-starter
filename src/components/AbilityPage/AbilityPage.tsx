import * as React from 'react';
import './abilityPage.scss';
import { IArticleModel } from 'src/models';
import { match } from 'react-router';
import BreadcrumbsTemplate from '../template/breadcrumbs/breadcrumbs';
import FullScreenCont from '../template/fullScreenCont/fullSreenCont';
import ProjectContainer from '../ProjectPage/projectContainer/ProjectContainer';

export default class AbilityPageComp extends React.Component<IAbilityPageProps, any> {
    constructor(props: any) {
        super(props);       
    }

    public render() {
        const { abilities } = this.props;
        return (
            <main>
                <FullScreenCont>        
                    <img className="headerImage" src="../images/backgroundMainPages/news.jpg" alt=""/>
                </FullScreenCont>
                <BreadcrumbsTemplate match={this.props.match}/>
                <ProjectContainer content={abilities}/>
            </main>
        );
    }
}

interface IAbilityPageProps {
    abilities: Array<IArticleModel>,
    match: match<void>
}
