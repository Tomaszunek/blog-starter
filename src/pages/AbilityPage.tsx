import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { ArticleActions } from '../actions';
import { ArticleFiltes } from '../models';
import { IRootState, RootState } from '../reducers';
import { omit } from '../utils';
import '../App.scss';
import MainPageComp from 'src/components/MainPage/MainPageComp';

const FILTER_VALUES = (Object.keys(ArticleFiltes.Filter) as 
  (keyof typeof ArticleFiltes.Filter)[]).map(
  (key) => ArticleFiltes.Filter[key]
);


export namespace AbilityPage {
  export interface IProps extends RouteComponentProps<void> {
    articles: RootState.ArticleState;
    products: RootState.ProductState;
    actions: ArticleActions;
    filter: ArticleFiltes.Filter;
  }
}

@connect(
  (state: IRootState, ownProps): Pick<AbilityPage.IProps, 'articles' | 'products' | 'filter'> => {
    const hash = ownProps.location && ownProps.location.hash.replace('#', '');
    const filter = FILTER_VALUES.find((value) => value === hash) || ArticleFiltes.Filter.SHOW_ALL;
    return { articles: state.articles, products: state.products, filter };
  },  
  (dispatch: Dispatch): Pick<AbilityPage.IProps, 'actions'> => ({
    actions: bindActionCreators(omit(ArticleActions, 'Type'), dispatch)
  })
)

export default class AbilityPage extends React.Component<AbilityPage.IProps> {     
  public render() {
    const { articles } = this.props; 
    return (
      <MainPageComp articles={articles}/>
    );
  }
}


