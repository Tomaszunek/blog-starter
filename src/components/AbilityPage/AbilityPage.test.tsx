import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import AbilityPage from './AbilityPage'
import { IArticleModel } from 'src/models';

describe('Error Page', () => {
    describe('Component should render right', () => {
        let app:any;
        const testMatch:any = {
            path: '',
            params: {},
            url: ''
        };
        const abilities:Array<IArticleModel> = [{
            id: 1,
            type: "ability",
            articleType: "ability",
            body: "body",
            createdAt: new Date(),
            image: "images/placeholder/abc.jpg",
            name: "woda",
            slug: "woda",
            updatedAt: new Date(),
        },{
            id: 2,
            type: "ability",
            articleType: "ability",
            body: "body",
            createdAt: new Date(),
            image: "images/placeholder/abc.jpg",
            name: "woda",
            slug: "woda",
            updatedAt: new Date(),
        }]
        beforeEach(() => {
            app = mount(<AbilityPage abilities={abilities} match={testMatch}/>);
        })
        it('render correctly snapshot component', () => {
            const tree = renderer
            .create(<AbilityPage abilities={abilities} match={testMatch}/>)
            .toJSON();
            expect(tree).toMatchSnapshot();
        })
        it('component should contains compoments', () => {
            expect(app.exists('.breadcrumbs')).toBeTruthy();
            expect(app.exists('.fullScreenCont')).toBeTruthy();
            expect(app.exists('.projectContainer')).toBeTruthy();
        })
        it('should render proper count item', () => {
            expect(app.find('.projectContainer').children()).toHaveLength(abilities.length);
        })
        
    });
  });