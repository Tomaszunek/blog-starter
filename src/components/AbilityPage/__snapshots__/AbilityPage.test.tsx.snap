// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page Component should render right render correctly snapshot component 1`] = `
<main>
  <div
    className="fullScreenCont"
  >
    <img
      alt=""
      className="headerImage"
      src="../images/backgroundMainPages/news.jpg"
    />
  </div>
  <ul
    className="breadcrumbs"
  >
    <li
      className="home"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      className="current"
    >
      <a />
    </li>
  </ul>
  <div
    className="projectContainer "
  >
    <a
      className="projectTile ability"
      href={undefined}
    >
      <img
        alt=""
        src="/images/projects/images/placeholder/abc.jpg"
      />
      <div
        className="categoryTile"
      >
        <span>
          woda
        </span>
      </div>
      <p>
        body
      </p>
    </a>
    <a
      className="projectTile ability"
      href={undefined}
    >
      <img
        alt=""
        src="/images/projects/images/placeholder/abc.jpg"
      />
      <div
        className="categoryTile"
      >
        <span>
          woda
        </span>
      </div>
      <p>
        body
      </p>
    </a>
  </div>
</main>
`;
